### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by casperdcl**
- [x] add codecov
- [x] add coveralls
- [ ] add python tests
    + [x] pSTIR
    + [ ] pGadgetron
    + [ ] other?
- [ ] add c tests
- [ ] add other (matlab) tests?
- [x] add travis
    - [x] fix travis (upstream SuperBuild issue, https://github.com/CCPPETMR/SIRF-SuperBuild/issues/57)

### CODE REVIEW

**Comment by KrisThielemans** [comment]
@evgueni-ovtchinnikov made the Python tests nosests compatible, while still being able to run them as we do now. It's on the `devel` branch, commit 1efd811ca6dd1a35c0a95b80e0a445b2bd688c27. @casperdcl would this be ok to merge to this branch?

For MATLAB, I repeat text from my email here for easy reference: recent MATLAB is able to output Cobertura xml (if we’d use unittest). https://uk.mathworks.com/help/matlab/ref/matlab.unittest.plugins.codecoverage.html?s_tid=doc_ta

**Comment by casperdcl** [comment]
Hi @KrisThielemans (CC: @evgueni-ovtchinnikov) I've merged `devel` into this branch (`coverage`), and switched to using `ctest`. It's mergeable into `master` as-is, since the failures are the same as what's already on `master`

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then curl -0 https://cmake.org/files/v3.7/cmake-3.7.2-Linux-x86_64.tar.gz -o cmake.tar.gz; fi
+  - tar xzf cmake.tar.gz
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then mv cmake-*/CMake.app/Contents/* cmake-*; fi
+
+install:
+  - #pip install --upgrade pip setuptools wheel
+  - pip install --only-binary=numpy,scipy,matplotlib numpy scipy matplotlib
+  - pip install nose nose-timer codecov coveralls
+  - cd ..
+  - git clone https://github.com/CCPPETMR/SIRF-SuperBuild --recursive -b v0.9.2
```
Comment (KrisThielemans) [inline .travis.yml]: need to delete the 0.9.2 checkout and stick to master. (it will remove some errors)

Comment (casperdcl) [reply .travis.yml]: done

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_HDF5=Off -DUSE_SYSTEM_SWIG=Off -DPYTHON_LIBRARY=$(python-config --prefix)/lib/libpython2.7.dylib -DPYTHON_INCLUDE_DIR=$(python-config --prefix)/include/python2.7" CC=gcc CXX=g++
+  - os: osx
+    language: generic
+    env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_HDF5=OFF -DUSE_SYSTEM_SWIG=On -DDEVEL_BUILD=ON -DPYTHON_LIBRARY=$(python-config --prefix)/lib/libpython2.7.dylib -DPYTHON_INCLUDE_DIR=$(python-config --prefix)/include/python2.7 "
+  - os: osx
+    language: generic
+    env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_SWIG=On -DUSE_SYSTEM_HDF5=Off -DUSE_SYSTEM_FFTW3=OFF -DDEVEL_BUILD=ON -DPYTHON_LIBRARY=$(python-config --prefix)/lib/libpython2.7.dylib -DPYTHON_INCLUDE_DIR=$(python-config --prefix)/include/python2.7" CC=gcc CXX=g++
+env:
+  global:
+    - BUILD_FLAGS="-DCMAKE_BUILD_TYPE=Release -DSIRF_TAG=$TRAVIS_COMMIT -DGADGETRON_TAG=e7eb430"
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply .travis.yml]: ok removed gadgetron. Don't want to set the URL here as it $PWD might not be correct at this point and also $PWD/../SIRF will not (yet) exist. Better to set the URL explicitly later just after cloning

[the remark this answers is not in the document]
Comment (KrisThielemans) [reply .travis.yml]: ok. would you mind adding a comment on it in the file? It got me confused for a while

**Comment by KrisThielemans** [comment]
I guess the tic boxes can be ticked. we just rely on whatever is done by ctest now, correct?

We do now 2 .travis files. here and the Superbuild. That's going to be require a lot of discipline... any ideas on how to avoid/reduce the duplication? we could create some shell scripts for the major steps, but the question is obviously in which repo they'd have to be and how to get at them in the other repo.

**Comment by casperdcl** [comment]
I was aiming to get this merged first, then sort out the mac build issues so travis fully passes, then get SIRF-SuperBuild:system-build merged in so we can all start using docker/*.sh scripts more, including for travis and VMs.


Happy to hear any other suggestions.

**Comment by evgueni-ovtchinnikov** [comment]
nosetests: error: no such option: --with-timer

tests run ok if this option is removed

**Comment by casperdcl** [comment]
`--with-timer` requires `pip install nose-timer`

**Comment by casperdcl** [comment]
you could always `pip install --user nose-timer` or use a `virtualenv`

**Comment by evgueni-ovtchinnikov** [comment]
thanks!

we need to have it in SuperBuild

**Comment by KrisThielemans** [comment]
the superbulid currently doesn't install python packages. I think that is too complicated (too many python variations around).

**Comment by evgueni-ovtchinnikov** [comment]
then better make do without timer?

**Comment by evgueni-ovtchinnikov** [comment]
just discovered that we do not have nosetests on VM

**Comment by evgueni-ovtchinnikov** [comment]
forgot to add: consequently, ctest fails on VM

**Comment by paskino** [comment]
Should I add nose-timer to the VM?

**Comment by casperdcl** [comment]
quick update: we're not using `nose-timer` any more (I suppose `ctests`'s coarse time is enough)

**Pull request merged**